lightning/src/chain/channelmonitor.rs

```diff
@@ -2902,9 +2954,9 @@ impl<Signer: Sign> ChannelMonitorImpl<Signer> {
 
 			if tx.input.len() == 1 {
 				// Assuming our keys were not leaked (in which case we're screwed no matter what),
-				// commitment transactions and HTLC transactions will all only ever have one input,
-				// which is an easy way to filter out any potential non-matching txn for lazy
-				// filters.
+				// commitment transactions and HTLC transactions will all only ever have one input
+				// (except for HTLC transactions for channels with anchor outputs), which is an easy
+				// way to filter out any potential non-matching txn for lazy filters.
```

Comment:
Wait if we filter out any transaction with a number of inputs superior to 1 in this conditional flow L2955 - L3008, and we call `check_spend_counterparty_htlc` L2999 aiming to cover the revoked post-anchor batched HTLC transaction from our counterparty and be supposed to be safe against the vulnerability described L2660 ? Note,
this is the only callsite for `check_spend_counterparty_htlc`. Am I correct we're wrong here ?